### ISSUE TITLE
<core-list> card layout problems

### ISSUE THREAD

**Issue by WrinklyNinja** [loot/loot#426]
This inherits from #364, I decided to open a new issue because the cause is different from the old one.

There are two symptoms:

1. When jumping to cards near the bottom of the list, the space above them will shrink, until they hit `scrollTop`. More description and screenshots [here](http://forums.bethsoft.com/topic/1516193-rel-loot-thread-18/?p=23949021)
2. When jumping to new cards using the sidebar navigation, cards can be repositioned such that the top card(s) (which may be visible only after scrolling)  overlap with the general information card.

**Comment by WrinklyNinja** [comment]
*This comment has been copied from the previous issue.*

I can reliably replicate symptom 2 by jumping to `ThePitt.esm`'s card using the sidebar. 

The problem seems to be that in `core-list.html` in `_resetIndex()`, the `_scrollTop` variable is being set to a nonsense value on the line

```
 this._scrollTop = this.setScrollTop(this._aboveSize + (index / this._rowFactor) * this._physicalAverage);
```

It's going wrong because the `_physicalAverage` is inaccurate. In this case, it's too small. This means that the `_physicalOffset` is set to too small a value on the next line, and so when the actual physical sizes are subtracted from it in `_positionItems()`, the result turns negative, and everything is transformed so that plugin cards overlap the summary card.

I tried setting the correct scroll top value, but the browser will clamp the value so that it is at most the total scroll height - the visible area of the element. So it seems that using `_physicalAverage` doesn't actually matter.

**Comment by WrinklyNinja** [comment]
Symptom 2 seems to be restricted to when the list is only a little bit taller than the viewport, there must be a bug with the scrolling / positioning calculations. I can replicate it on Polymer's `<core-list>` demo page.

Symptom 1 can also be replicated on the demo page.

**Comment by WrinklyNinja** [comment]
After some more thought, it appears the two symptoms have exactly the same cause, with symptom 2 being the result of the special case where the list is very short. I've filed issue 91 on the Polymer/core-list issue tracker detailing it.

**Comment by WrinklyNinja** [comment]
OK, I fixed the cards overlapping by clamping `<core-list>._physicalOffset` to non-negative values.

The shrinking list height seems to be linked to `<core-list>._physicalAverage` decreasing due to the average size of each card increasing each time `<core-list>.scrollToItem()` is called for the last item.
